tests/unit/assessment/test_assessment.py

```diff
@@ -2514,3 +2518,134 @@ def test_list_all_pipeline_with_conf_spn_secret_avlb(mocker):
     assert result_set[0].get("application_id") == "Hello, World!"
     assert result_set[0].get("tenant_id") == "directory_12345"
     assert result_set[0].get("storage_account") == "newstorageacct"
+
+
+def test_job_crawler_with_no_owner_should_have_empty_creator_name():
+    sample_jobs = [
+        BaseJob(
+            created_time=1694536604319,
+            creator_user_name=None,
+            job_id=536591785949415,
+            settings=JobSettings(
+                compute=None,
+                continuous=None,
+                tasks=[
+                    Task(
+                        task_key="Ingest",
+                        existing_cluster_id="0807-225846-avon493",
+                        notebook_task=NotebookTask(
+                            notebook_path="/Users/foo.bar@databricks.com/Customers/Example/Test/Load"
+                        ),
+                        timeout_seconds=0,
+                    )
+                ],
+                timeout_seconds=0,
+            ),
+        )
+    ]
+
+    sample_clusters = [
+        ClusterDetails(
+            autoscale=AutoScale(min_workers=1, max_workers=6),
+            spark_context_id=5134472582179566666,
+            spark_env_vars=None,
+            spark_version="13.3.x-cpu-ml-scala2.12",
+            cluster_id="0807-225846-avon493",
+            cluster_source=ClusterSource.JOB,
+        )
+    ]
+    ws = Mock()
+    mockbackend = MockBackend()
+    ws.jobs.list.return_value = sample_jobs
+    ws.clusters.list.return_value = sample_clusters
+    JobsCrawler(ws, mockbackend, "ucx").snapshot()
+    result = mockbackend.rows_written_for("hive_metastore.ucx.jobs", "append")
+    assert result == [JobInfo(job_id="536591785949415", job_name=None, creator="", success=1, failures="[]")]
```

Comment:
Can we do None for creator name everywhere, instead of empty strings? It translates into NULL, which has easier semantics

Reply:
I've went with this pattern at first ... But it didn't worked for most of the objects because classes needs to be changed from this:

```
class JobInfo:
    job_id: str
    job_name: str
    creator: str
    success: int
    failures: str
```

to this

```
class JobInfo:
    job_id: str
    job_name: str
    success: int
    failures: str
    creator: str: None

```
Which is a breaking change as it changes column order ...

Reply:
Even using a class like this:

class JobInfo:
    job_id: str
    job_name: str
    creator: str | None
    success: int
    failures: str

is breaking because creator goes from NOT NULL to NULLABLE, but it won't require re-creation of existing ucx deployment if there are no objects with unknown owners

Reply:
> Which is a breaking change as it changes column order ...

our insert statement generation is aware of the actual column order